app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (NotWearingMaskError,
+                        NotVaccinatedError,
+                        OutdatedVaccineError)
+
+
+def go_to_cafe(friends, cafe) -> str:
+    total = 0
+
+    for i in range(len(friends)):
+        try:
+            cafe.visit_cafe(friends[i])
+        except (NotVaccinatedError, OutdatedVaccineError):
+            return 'All friends should be vaccinated'
+        except NotWearingMaskError:
+            total += 1
+
+    if total:
+        return f'Friends should buy {total} masks'
+    else:
+        return f'Friends can go to {cafe.name}'
```

Comment:
Before returning a success message and in case of catching general `Exception`, I would advice counting the number of visitors who can successfully visit the cafe and return the success message only in case this number equals to `len(friends)`. It would require additional refinement to define what message should be returned in case of unforeseen errors, but it could be beneficial for the consumer.